ext/CEGUI/IrrlichtBaWTexture.cpp

```diff
@@ -15,35 +16,99 @@ IrrlichtBaWTexture::~IrrlichtBaWTexture()
+{
+    asset::ICPUImage::SCreationParams imgInfo;
+    imgInfo.format = colorAttachmentFormat;
+    imgInfo.type = asset::ICPUImage::ET_2D;
+    imgInfo.extent.width = width;
+    imgInfo.extent.height = height;
+    imgInfo.extent.depth = 1u;
+    imgInfo.mipLevels = 1u;
+    imgInfo.arrayLayers = 1u;
+    imgInfo.samples = asset::ICPUImage::ESCF_1_BIT;
+    imgInfo.flags = static_cast<asset::IImage::E_CREATE_FLAGS>(0u);
+
+    auto image = asset::ICPUImage::create(std::move(imgInfo));
+    const auto texelFormatBytesize = getTexelOrBlockBytesize(image->getCreationParameters().format);
+
+    core::smart_refctd_ptr<asset::ICPUBuffer> texelBuffer = core::make_smart_refctd_ptr<asset::CCustomAllocatorCPUBuffer<core::null_allocator<uint8_t>>>(image->getImageDataSizeInBytes(), data, core::adopt_memory);
+    auto regions = core::make_refctd_dynamic_array<core::smart_refctd_dynamic_array<asset::ICPUImage::SBufferCopy>>(1u);
+    asset::ICPUImage::SBufferCopy& region = regions->front();
+
+    region.imageSubresource.mipLevel = 0u;
+    region.imageSubresource.baseArrayLayer = 0u;
+    region.imageSubresource.layerCount = 1u;
+    region.bufferOffset = 0u;
+    region.bufferRowLength = image->getCreationParameters().extent.width;
+    region.bufferImageHeight = 0u;
+    region.imageOffset = { 0u, 0u, 0u };
+    region.imageExtent = image->getCreationParameters().extent;
+
+    image->setBufferAndRegions(std::move(texelBuffer), regions);
+
+    asset::ICPUImageView::SCreationParams imgViewInfo;
+    imgViewInfo.image = std::move(image);
+    imgViewInfo.format = colorAttachmentFormat;
+    imgViewInfo.viewType = asset::IImageView<asset::ICPUImage>::ET_2D;
+    imgViewInfo.flags = static_cast<asset::ICPUImageView::E_CREATE_FLAGS>(0u);
+    imgViewInfo.subresourceRange.baseArrayLayer = 0u;
+    imgViewInfo.subresourceRange.baseMipLevel = 0u;
+    imgViewInfo.subresourceRange.layerCount = imgInfo.arrayLayers;
+    imgViewInfo.subresourceRange.levelCount = imgInfo.mipLevels;
+
+    auto imageView = asset::ICPUImageView::create(std::move(imgViewInfo));
+    auto gpuImageView = driver->getGPUObjectsFromAssets(&imageView.get(), &imageView.get() + 1)->front();
+
+    return std::move(gpuImageView);
+}
+
+irr::video::IFrameBuffer* IrrlichtBaWTexture::createFrameBuffer(const void* data, asset::E_FORMAT colorAttachmentFormat, size_t width, size_t height)
+{
+    auto gpuImageViewColorBuffer = createAndFillAttachement(driver, data, colorAttachmentFormat, width, height);
+
+    auto frameBuffer = driver->addFrameBuffer();
+    frameBuffer->attach(textureColorAttachment, std::move(gpuImageViewColorBuffer));
+
+    return frameBuffer;
+}
+
+irr::video::IFrameBuffer* IrrlichtBaWTexture::createFrameBuffer(core::smart_refctd_ptr<video::IGPUImageView> gpuImageView)
+{
+    auto frameBuffer = driver->addFrameBuffer();
+    frameBuffer->attach(textureColorAttachment, std::move(gpuImageView));
+
+    return frameBuffer;
+}
```

Comment:
you don't need to use the framebuffer at all

Reply:
adjusted in following commit